[harfbuzz] build failure

**Host Environment**
- OS: Windows10
- Compiler:. MSVC (VS2019)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install harfbuzz:x64-windows-static-md`

**Failure logs** 
```
.\vcpkg install harfbuzz:x64-windows-static-md
Computing installation plan...
The following packages will be built and installed:
    harfbuzz[core]:x64-windows-static-md -> 2.7.4
Detecting compiler hash for triplet x64-windows-static-md...
Could not locate cached archive: C:\Users\dev-user\AppData\Local\vcpkg\archives\1f\1f35d44d702c88f19d8772179a3342b127d4f716.zip
Starting package 1/1: harfbuzz:x64-windows-static-md
Building package harfbuzz[core]:x64-windows-static-md...
-- Using community triplet x64-windows-static-md. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: \\VBoxSvr\Shared\vcpkg\triplets\community\x64-windows-static-md.cmake
-- Using cached S:/vcpkg/downloads/harfbuzz-harfbuzz-7236c7e29cef1c2d76c7a284c5081ff4d3aa1127.tar.gz
-- Extracting source S:/vcpkg/downloads/harfbuzz-harfbuzz-7236c7e29cef1c2d76c7a284c5081ff4d3aa1127.tar.gz
-- Applying patch 0002-fix-uwp-build.patch
-- Using source at S:/vcpkg/buildtrees/harfbuzz/src/f4d3aa1127-a820a32874.clean
-- Configuring x64-windows-static-md
-- Using msys root at S:/vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x64-windows-static-md-dbg
CMake Error at //VBoxSvr/Shared/vcpkg/scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: S:/vcpkg/downloads/tools/python/python-3.9.0-x64/python.exe S:/vcpkg/downloads/tools/meson/meson-f5871f434a5b768ad9fcafa797a6db0286421842/meson.py -Dicu=disabled -Dgraphite=disabled -Dglib=disabled -Dfreetype=enabled -Dgobject=disabled -Dcairo=disabled -Dfontconfig=disabled -Dintrospection=disabled -Ddocs=disabled -Dtests=disabled -Dbenchmark=disabled --buildtype plain --backend ninja --wrap-mode nodownload --native S:/vcpkg/buildtrees/harfbuzz/meson-nativ-x64-windows-static-md.log --default-library static --libdir lib --native S:/vcpkg/buildtrees/harfbuzz/meson-nativ-x64-windows-static-md-debug.log -Ddebug=true --prefix S:/vcpkg/packages/harfbuzz_x64-windows-static-md/debug --includedir ../include -Dcmake_prefix_path=['S:/vcpkg/installed/x64-windows-static-md/debug','S:/vcpkg/installed/x64-windows-static-md','S:/vcpkg/installed/x64-windows-static-md/share'] S:/vcpkg/buildtrees/harfbuzz/src/f4d3aa1127-a820a32874.clean
    Working Directory: S:/vcpkg/buildtrees/harfbuzz/x64-windows-static-md-dbg
    Error code: 1
    See logs for more information:
      S:\vcpkg\buildtrees\harfbuzz\config-x64-windows-static-md-dbg-out.log

Call Stack (most recent call first):
  //VBoxSvr/Shared/vcpkg/scripts/cmake/vcpkg_configure_meson.cmake:482 (vcpkg_execute_required_process)
  ports/harfbuzz/portfile.cmake:20 (vcpkg_configure_meson)
  //VBoxSvr/Shared/vcpkg/scripts/ports.cmake:131 (include)


Error: Building package harfbuzz:x64-windows-static-md failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: harfbuzz:x64-windows-static-md
  Vcpkg version: 2021-01-13-768d8f95c9e752603d2c5901c7a7c7fbdb08af35

Additionally, attach any relevant sections from the log files above.
```
The mentioned log file:
```
DEPRECATION: c_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: c_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
WARNING: Recommend using either -Dbuildtype or -Doptimization + -Ddebug. Using both is redundant since they override each other. See: https://mesonbuild.com/Builtin-options.html#build-type-options
The Meson build system
Version: 0.56.99
Source dir: \\VBoxSvr\Shared\vcpkg\buildtrees\harfbuzz\src\f4d3aa1127-a820a32874.clean
Build dir: \\VBoxSvr\Shared\vcpkg\buildtrees\harfbuzz\x64-windows-static-md-dbg
Build type: native build
Project name: harfbuzz
Project version: 2.7.4
C compiler for the host machine: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe (msvc 19.28.29334 "Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29334 for x64")
C linker for the host machine: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/link.exe link 14.28.29334.0
C++ compiler for the host machine: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/cl.exe (msvc 19.28.29334 "Microsoft (R) C/C++ Optimizing Compiler Version 19.28.29334 for x64")
C++ linker for the host machine: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29333/bin/Hostx64/x64/link.exe link 14.28.29334.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments /utf-8: YES 
Compiler for C++ supports link arguments -Bsymbolic-functions: NO 
Compiler for C++ supports arguments -fno-exceptions: NO 
Compiler for C++ supports arguments -fno-rtti: NO 
Compiler for C++ supports arguments -fno-threadsafe-statics: NO 
Compiler for C++ supports arguments -fvisibility-inlines-hidden: NO 
Library m found: NO
Found pkg-config: S:/vcpkg/downloads/tools/msys2/50864d9fae363d6e/mingw32/bin/pkg-config.exe (0.29.2)
Run-time dependency freetype2 found: YES 23.4.17
Dependency glib-2.0 skipped: feature glib disabled
Dependency gobject-2.0 skipped: feature gobject disabled
Dependency fontconfig skipped: feature fontconfig disabled
Dependency graphite2 skipped: feature graphite disabled
Compiler for C++ supports arguments -Wno-non-virtual-dtor: NO 
WARNING: msvc does not support C++11; attempting best effort; setting the standard to C++14
Has header "unistd.h" : NO 
Has header "sys/mman.h" : NO 
Has header "stdbool.h" : YES 
Checking for function "atexit" : NO 
Checking for function "mprotect" : NO 
Checking for function "sysconf" : NO 
Checking for function "getpagesize" : NO 
Checking for function "mmap" : NO 
Checking for function "isatty" : YES 
Checking for function "FT_Get_Var_Blend_Coordinates" with dependency freetype2: YES 
Checking for function "FT_Set_Var_Blend_Coordinates" with dependency freetype2: YES 
Checking for function "FT_Done_MM_Var" with dependency freetype2: YES 
Checking if "Intel atomics" links: NO 
Checking if "Solaris atomic ops" links: NO 
Program gen-hb-version.py found: YES (S:\vcpkg\downloads\tools\python\python-3.9.0-x64\python.exe \\VBoxSvr\Shared\vcpkg\buildtrees\harfbuzz\src\f4d3aa1127-a820a32874.clean\src\gen-hb-version.py)
WARNING: Project targeting '>= 0.47.0' but tried to use feature introduced in '0.50.0': install arg in configure_file.
Configuring hb-version.h with command

\\VBoxSvr\Shared\vcpkg\buildtrees\harfbuzz\src\f4d3aa1127-a820a32874.clean\src\meson.build:1:0: ERROR: Running configure command failed.

Traceback (most recent call last):
  File "\\VBoxSvr\Shared\vcpkg\buildtrees\harfbuzz\src\f4d3aa1127-a820a32874.clean\src\gen-hb-version.py", line 36, in <module>
    shutil.copyfile (OUTPUT, os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT)))
  File "shutil.py", line 241, in copyfile
shutil.SameFileError: '\\\\VBoxSvr\\Shared\\vcpkg\\buildtrees\\harfbuzz\\x64-windows-static-md-dbg\\src\\hb-version.h' and '\\\\VBoxSvr\\Shared\\vcpkg\\buildtrees\\harfbuzz\\src\\f4d3aa1127-a820a32874.clean\\src\\hb-version.h' are the same file
```

I assme the error is either
```
WARNING: Project targeting '>= 0.47.0' but tried to use feature introduced in '0.50.0': install arg in configure_file.
```
or the last lines about these files being the same. Idk what files these are, so I don't just want to blindly start deleting either of them to see if that resolves the issue :thinking: 

**Additional context**

I am using the latest vcpkg version available as of now: e2ecca095c77180ab0114600f78b557ab6c99fc2

--- a/ports/harfbuzz/0001-circumvent-samefile-error.patch
+++ b/ports/harfbuzz/0001-circumvent-samefile-error.patch
@@ -1,0 +1,43 @@
+diff --git a/src/gen-harfbuzzcc.py b/src/gen-harfbuzzcc.py
+index b25bcc7..97bf2ab 100644
+--- a/src/gen-harfbuzzcc.py
++++ b/src/gen-harfbuzzcc.py
+@@ -15,4 +15,8 @@ with open (OUTPUT, "wb") as f:
+        f.write ("".join ('#include "{}"\n'.format (os.path.basename (x)) for x in sources if x.endswith (".cc")).encode ())
+ 
+ # copy it also to src/
+-shutil.copyfile (OUTPUT, os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT)))
++src = OUTPUT
++dst = os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT))
++# Avoid SameFileError
++if not os.path.samefile(src, dst):
++    shutil.copyfile (src, dst)
+diff --git a/src/gen-hb-version.py b/src/gen-hb-version.py
+index 5ec2024..59d4754 100644
+--- a/src/gen-hb-version.py
++++ b/src/gen-hb-version.py
+@@ -33,4 +33,8 @@ with open (INPUT, "r", encoding='utf-8') as template:
+                        .encode ())
+ 
+ # copy it also to src/
+-shutil.copyfile (OUTPUT, os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT)))
++src = OUTPUT
++dst = os.path.join(CURRENT_SOURCE_DIR, os.path.basename (OUTPUT))
++# Avoid SameFileError if src and dst is the same
++if not os.path.samefile(src, dst):
++    shutil.copyfile (src, dst)
+diff --git a/src/gen-ragel-artifacts.py b/src/gen-ragel-artifacts.py
+index b60ec3b..288e224 100644
+--- a/src/gen-ragel-artifacts.py
++++ b/src/gen-ragel-artifacts.py
+@@ -22,4 +22,8 @@ hh = rl.replace ('.rl', '.hh')
+ subprocess.Popen ([ragel, '-e', '-F1', '-o', hh, rl], cwd=outdir).wait ()
+ 
+ # copy it also to src/
+-shutil.copyfile (os.path.join (outdir, hh), os.path.join (CURRENT_SOURCE_DIR, hh))
++src = os.path.join (outdir, hh)
++dst = os.path.join (CURRENT_SOURCE_DIR, hh)
++# Avoid SameFileError
++if not os.path.samefile(src, dst):
++    shutil.copyfile (src, dst)
+

--- a/ports/harfbuzz/portfile.cmake
+++ b/ports/harfbuzz/portfile.cmake
@@ -5,6 +5,11 @@ vcpkg_from_github(
     SHA512 d231a788ea4e52231d4c363c1eca76424cb82ed0952b5c24d0b082e88b3dddbda967e7fffe67fffdcb22c7ebfbf0ec923365eb4532be772f2e61fa7d29b51998
     HEAD_REF master
     PATCHES
+		# This patch is a workaround that is needed until the following issues are resolved upstream:
+		# - https://github.com/mesonbuild/meson/issues/8375
+		# - https://github.com/harfbuzz/harfbuzz/issues/2870
+		# Details: https://github.com/microsoft/vcpkg/issues/16262
+		0001-circumvent-samefile-error.patch
         0002-fix-uwp-build.patch
 )
 

--- a/ports/harfbuzz/vcpkg.json
+++ b/ports/harfbuzz/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "harfbuzz",
   "version-string": "2.7.4",
+  "port-version": 1,
   "description": "HarfBuzz OpenType text shaping engine",
   "homepage": "https://github.com/harfbuzz/harfbuzz",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2378,7 +2378,7 @@
     },
     "harfbuzz": {
       "baseline": "2.7.4",
-      "port-version": 0
+      "port-version": 1
     },
     "hayai": {
       "baseline": "2019-08-10",

--- a/versions/h-/harfbuzz.json
+++ b/versions/h-/harfbuzz.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e985af9b39fb57ee491c1a773c43334665ddb3d3",
+      "version-string": "2.7.4",
+      "port-version": 1
+    },
+    {
       "git-tree": "bb9ea75cd35a35e57fb0bf79ff78818c95148fcf",
       "version-string": "2.7.4",
       "port-version": 0

